src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationIncrementalImporter.java

```diff
@@ -0,0 +1,111 @@
+    private final int geneticProfileId;
+    private final List<Integer> fileOrderedSampleList;
+    private final List<Integer> extendedOrderedSampleList;
+    private final DaoGeneticAlteration daoGeneticAlteration = DaoGeneticAlteration.getInstance();
+    private final HashMap<Integer, HashMap<Integer, String>> geneticAlterationMap;
+
+    public GeneticAlterationIncrementalImporter(
+            int geneticProfileId,
+            List<Integer> fileOrderedSampleList
+    ) throws DaoException {
+        this.geneticProfileId = geneticProfileId;
+        this.geneticAlterationMap = daoGeneticAlteration.getGeneticAlterationMapForEntityIds(geneticProfileId, null);
+        this.fileOrderedSampleList = fileOrderedSampleList;
+
+        ArrayList <Integer> savedOrderedSampleList = DaoGeneticProfileSamples.getOrderedSampleList(this.geneticProfileId);
+        int initialOrderSampleListSize = savedOrderedSampleList.size();
+        geneticAlterationMap.forEach((geneticEntityId, sampleToValue) -> {
+            if (sampleToValue.size() != initialOrderSampleListSize) {
+                throw new IllegalStateException("Number of samples ("
+                        + sampleToValue.size() + ") for genetic entity with id "
+                        + geneticEntityId + " does not match with the number in the inital sample list ("
+                        + initialOrderSampleListSize + ").");
+            }
+        });
+        // add all new sample ids at the end
+        this.extendedOrderedSampleList = new ArrayList<>(savedOrderedSampleList);
+        List<Integer> newSampleIds = this.fileOrderedSampleList.stream().filter(sampleId -> !savedOrderedSampleList.contains(sampleId)).toList();
+        this.extendedOrderedSampleList.addAll(newSampleIds);
+        DaoGeneticProfileSamples.deleteAllSamplesInGeneticProfile(this.geneticProfileId);
+        this.geneticAlterationImporter = new GeneticAlterationImporterImpl(geneticProfileId, extendedOrderedSampleList);
+    }
+
+    @Override
+    public boolean store(String[] values, CanonicalGene gene, String geneSymbol) throws DaoException {
+        int geneticEntityId = gene.getGeneticEntityId();
+        daoGeneticAlteration.deleteAllRecordsInGeneticProfile(geneticProfileId, geneticEntityId);
+        String[] expandedValues = extendValues(geneticEntityId, values);
+        return geneticAlterationImporter.store(expandedValues, gene, geneSymbol);
+    }
+
+    @Override
+    public boolean store(int geneticEntityId, String[] values) throws DaoException {
+        daoGeneticAlteration.deleteAllRecordsInGeneticProfile(geneticProfileId, geneticEntityId);
+        String[] expandedValues = extendValues(geneticEntityId, values);
+        return geneticAlterationImporter.store(geneticEntityId, expandedValues);
+    }
+
+    @Override
+    public void finalise() {
+        expandRemainingGeneticEntityTabDelimitedRowsWithBlankValue();
+        geneticAlterationImporter.finalise();
```

Comment:
why this last line?

Reply:
`geneticAlterationImporter.finalise()` does nothing atm. So removing the line won't change the behavior now. I wanted to call the finalise to notify the wrapped class to clean things, in case such logic appears there.

Let me know if you think we should remove the call.

Reply:
should we use `super.finalise()` instead? I think that would make more sense. Same for the other methods in this class like `store`

Reply:
@pieterlukasse please note `GeneticAlterationIncrementalImporter` is not a superclass of `GeneticAlterationImporterImpl`. I intentionally use composition over inheritance here to address the fragile base class problem.

Reply:
got it. I still think writing it in the conventional way through inheritance makes the code easier to follow.

Reply:
done https://github.com/cBioPortal/cbioportal-core/commit/d98392a291a3ace8afb137c3649ed3c272166fb5

Reply:
should `super.` be replaced by `this.` in a number of assignments?

Reply:
It can. It's a good question whether it should. I thought using super in the constructor communicates better that's an inherited field.

Reply:
here is the change https://github.com/cBioPortal/cbioportal-core/pull/47/commits/96acec541d3afec7ac19f189cb1cd0a471dbcafc